Automattic/jetpack
# My Jetpack: Refactor ConnectionScreen component

## Proposed changes:

This PR is preparatory work for further refactoring. It moves the sub-components of `ConnectScreen` into their own files in the `my-jetpack` package.

In more details:
- it moves the `useMyJetpackReturnToPage` hook into its own file
- it moves the `ConnectionScreenFooter ` component into its own file
- it moves the body of the `ConnectionScreen ` component into its own file
- it updates the TS settings

### Other information:


## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with this branch
- Verify that the connect screen looks the same and that you can complete the connection flow
<img width="500" alt="Screenshot 2024-03-21 at 9 42 58 AM" src="https://github.com/Automattic/jetpack/assets/1620183/8f39f23a-9d42-47bf-a7b6-d8fb9bde13a2">
